src/ripple/app/misc/HashRouter.cpp

```diff
@@ -19,85 +19,73 @@
 
 #include <BeastConfig.h>
 #include <ripple/app/misc/HashRouter.h>
-#include <ripple/basics/UptimeTimer.h>
-#include <map>
-#include <mutex>
 
 namespace ripple {
 
 auto
-HashRouter::emplace (uint256 const& index)
+HashRouter::emplace (uint256 const& key)
     -> std::pair<Entry&, bool>
 {
-    auto fit = mSuppressionMap.find (index);
+    auto iter = mSuppressionMap.find (key);
 
-    if (fit != mSuppressionMap.end ())
+    if (iter != mSuppressionMap.end ())
     {
+        mSuppressionMap.touch(iter);
         return std::make_pair(
-            std::ref(fit->second), false);
+            std::ref(iter->second), false);
     }
 
-    int elapsed = UptimeTimer::getInstance ().getElapsedSeconds ();
-    int expireCutoff = elapsed - mHoldTime;
-
     // See if any supressions need to be expired
-    auto it = mSuppressionTimes.begin ();
-
-    while ((it != mSuppressionTimes.end ()) && (it->first <= expireCutoff))
-    {
-        for(auto const& lit : it->second)
-            mSuppressionMap.erase (lit);
-        it = mSuppressionTimes.erase (it);
-    }
+    beast::expire(mSuppressionMap,
```

Comment:
I suspect that the `beast::` qualifier is not necessary, thanks to argument dependent lookup. I'm not 100% sure though, you might try it.